src/aof.c

```diff
@@ -698,41 +1260,40 @@ struct client *createAOFClient(void) {
     return c;
 }
 
-/* Replay the append log file. On success AOF_OK is returned,
+/* Replay an append log file. On success AOF_OK is returned,
  * otherwise, one of the following is returned:
  * AOF_OPEN_ERR: Failed to open the AOF file.
  * AOF_NOT_EXIST: AOF file doesn't exist.
  * AOF_EMPTY: The AOF file is empty (nothing to load).
  * AOF_FAILED: Failed to load the AOF file. */
-int loadAppendOnlyFile(char *filename) {
+int loadSingleAppendOnlyFile(char *filename) {
     struct client *fakeClient;
-    FILE *fp = fopen(filename,"r");
     struct redis_stat sb;
     int old_aof_state = server.aof_state;
     long loops = 0;
     off_t valid_up_to = 0; /* Offset of latest well-formed command loaded. */
     off_t valid_before_multi = 0; /* Offset before MULTI command loaded. */
-    int ret;
+    off_t last_progress_report_size = 0;
+    int ret = C_OK;
 
+    sds aof_filepath = makePath(server.aof_dirname, filename);
+    FILE *fp = fopen(aof_filepath, "r");
     if (fp == NULL) {
         int en = errno;
-        if (redis_stat(filename, &sb) == 0) {
-            serverLog(LL_WARNING,"Fatal error: can't open the append log file for reading: %s",strerror(en));
+        if (redis_stat(aof_filepath, &sb) == 0) {
+            serverLog(LL_WARNING,"Fatal error: can't open the append log file %s for reading: %s", filename, strerror(en));
+            sdsfree(aof_filepath);
             return AOF_OPEN_ERR;
         } else {
-            serverLog(LL_WARNING,"The append log file doesn't exist: %s",strerror(errno));
+            serverLog(LL_WARNING,"The append log file %s doesn't exist: %s", filename, strerror(errno));
```

Comment:
```suggestion
            serverLog(LL_WARNING,"Failed to access the append log file %s: %s", filename, strerror(errno));
```

Updating the log message because the error may not necessarily be due to a non-existing file.

Reply:
No, the file must not exist here, because we ruled out this possibility in the if branch.

Reply:
and note we will return AOF_NOT_EXIST error code.

Reply:
what we know here is that both `fopen` and `fstat` failed.
i guess you could change
```diff
-if (redis_stat(aof_filepath, &sb) == 0) {
+if (redis_stat(aof_filepath, &sb) == 0 || errno!=ENOENT) {
```
then when you get here, you know for sure it doesn't exist.

Reply:
No , the `redis_stat` return 0 means that this file must exist.

Reply:
Yes, but does it mean that when it returns -1 the file must not exist? 
It could be a range of other errors.

Reply:
OK